src/polysolve/LinearSolverSympiler.cpp

```diff
@@ -0,0 +1,82 @@
+#ifdef POLYSOLVE_WITH_SYMPILER
+
+#include <polysolve/LinearSolverSympiler.hpp>
+#include <Eigen/Sparse>
+#include <Eigen/Core>
+
+namespace polysolve
+{
+    void LinearSolverSympiler::setSolverMode(int _solver_mode)
+    {
+        solver_mode = _solver_mode;
+    }
+
+    void LinearSolverSympiler::setParameters(const json &params)
+    {
+        if (params.contains("Sympiler"))
+        {
+            if (params["Sympiler"].contains("solver_mode"))
+            {
+                setSolverMode(params["Sympiler"]["mtype"].get<int>());
+            }
+        }
+    }
+
+    void LinearSolverSympiler::getInfo(json &params) const
+    {
+        if (factorize_status == 1)
+        {
+            params["factorize_info"] = "Success";
+        }
+        else
+        {
+            params["factorize_info"] = "Failure";
+        }
+    }
+
+    void LinearSolverSympiler::analyzePattern(const StiffnessMatrix &A, const int precond_num)
+    {
+        StiffnessMatrix A_copy = StiffnessMatrix(A);
```

Comment:
Is the copy necessary?

Reply:
Refactored so that this copy only happens when necessary